src/dashboard/infrastructure/integrations/site-kit.php

```diff
@@ -200,4 +240,37 @@ public function to_array(): array {
 	public function to_legacy_array(): array {
 		return $this->to_array();
 	}
+
+	/**
+	 * Parses the Site Kit configuration data.
+	 *
+	 * @return void
+	 */
+	public function parse_site_kit_data(): void {
+		$preload_paths       = \apply_filters( 'googlesitekit_apifetch_preload_paths', [] );
+		$actual_paths        = \array_intersect( $this->paths, $preload_paths );
+		$preloaded           = \array_reduce(
+			\array_unique( $actual_paths ),
+			'rest_preload_api_request',
+			[]
+		);
+		$modules_data        = $preloaded['/google-site-kit/v1/core/modules/data/list']['body'];
+		$modules_permissions = $preloaded['/google-site-kit/v1/core/user/data/permissions']['body'];
+		$is_authenticated    = $preloaded['/google-site-kit/v1/core/user/data/authentication']['body']['authenticated'];
+		foreach ( $modules_data as $module ) {
+			if ( $module['slug'] === 'analytics-4' ) {
+				$this->ga_module['owner']     = $module['owner'];
+				$this->ga_module['connected'] = $module['connected'];
+				if ( isset( $modules_permissions['googlesitekit_read_shared_module_data::["analytics-4"]'] ) ) {
+					$this->ga_module['permissions'] = $is_authenticated || $modules_permissions['googlesitekit_read_shared_module_data::["analytics-4"]'];
+				}
+			}
+			if ( $module['slug'] === 'search-console' ) {
+				$this->search_console_module['owner'] = $module['owner'];
+				if ( isset( $modules_permissions['googlesitekit_read_shared_module_data::["search-console"]'] ) ) {
+					$this->search_console_module['permissions'] = $is_authenticated || $modules_permissions['googlesitekit_read_shared_module_data::["search-console"]'];
```

Comment:
There's a lot of code duplication here begging for some abstraction, but I'm not sure if it's worth it, what do you think @thijsoo?

Maybe we can bypass the fact that `'connected'` is only a thing in GA and go ahead with some?